test/integration_test/applicationclone_test.go

```diff
@@ -10,34 +10,190 @@ import (
+	t *testing.T,
+	instanceID string,
+	appKey string,
+	additionalAppKeys []string,
+	cloneAppKey string,
+	cloneSuccessExpected bool,
+	cloneAllAppsExpected bool,
+) {
+	ctxs, err := schedulerDriver.Schedule(instanceID,
+		scheduler.ScheduleOptions{AppKeys: []string{appKey}})
+	require.NoError(t, err, "Error scheduling task")
+	require.Equal(t, 1, len(ctxs), "Only one task should have started")
+
+	err = schedulerDriver.WaitForRunning(ctxs[0], defaultWaitTimeout, defaultWaitInterval)
+	require.NoError(t, err, "Error waiting for app to get to running state")
+
+	cloneAppCtx := ctxs[0].DeepCopy()
 
-	cloneAppCtx := ctx.DeepCopy()
+	if len(additionalAppKeys) > 0 {
+		err = schedulerDriver.AddTasks(ctxs[0],
+			scheduler.ScheduleOptions{AppKeys: additionalAppKeys})
+		require.NoError(t, err, "Error scheduling additional apps")
+		err = schedulerDriver.WaitForRunning(ctxs[0], defaultWaitTimeout, defaultWaitInterval)
+		require.NoError(t, err, "Error waiting for additional apps to get to running state")
+	}
+
+	if cloneAllAppsExpected {
+		cloneAppCtx = ctxs[0].DeepCopy()
+	}
 
 	// Clone the app that was created and make sure the application clone task
 	// succeeds
 	cloneTaskCtx, err := schedulerDriver.Schedule("application-clone",
-		scheduler.ScheduleOptions{AppKeys: []string{"application-clone"}})
+		scheduler.ScheduleOptions{AppKeys: []string{cloneAppKey}})
 	require.NoError(t, err, "Error scheduling app clone task")
 	require.Equal(t, 1, len(cloneTaskCtx), "Only one task should have started")
-	err = schedulerDriver.WaitForRunning(cloneTaskCtx[0], defaultWaitTimeout, defaultWaitInterval)
-	require.NoError(t, err, "Error waiting for app clone task to get to running state")
+	timeout := defaultWaitTimeout
+	if !cloneSuccessExpected {
+		timeout = timeout / 4
+	}
+
+	err = schedulerDriver.WaitForRunning(cloneTaskCtx[0], timeout, defaultWaitInterval)
+	if cloneSuccessExpected {
+		require.NoError(t, err, "Error waiting for app clone task to get to running state")
 
-	// Make sure the cloned app is running
-	err = schedulerDriver.UpdateTasksID(cloneAppCtx, "app-clone-test-clone")
-	require.NoError(t, err, "Error updating task id for app clone context")
-	err = schedulerDriver.WaitForRunning(cloneAppCtx, defaultWaitTimeout, defaultWaitInterval)
-	require.NoError(t, err, "Error waiting for cloned app to get to running state")
+		// Make sure the cloned app is running
+		err = schedulerDriver.UpdateTasksID(cloneAppCtx, appKey+"-"+instanceID+"-dest")
+		require.NoError(t, err, "Error updating task id for app clone context")
+		err = schedulerDriver.WaitForRunning(cloneAppCtx, defaultWaitTimeout, defaultWaitInterval)
+		require.NoError(t, err, "Error waiting for cloned app to get to running state")
 
-	// Destroy the clone task and cloned add
-	destroyAndWait(t, []*scheduler.Context{cloneAppCtx, cloneTaskCtx[0]})
+		// Destroy the clone task and cloned app
+		err = schedulerDriver.UpdateTasksID(cloneAppCtx, appKey+"-"+instanceID+"-dest")
```

Comment:
Can we destroy without updating the taskID?

Reply:
Yeah don't need this, will remove it